cli/src/pcluster/templates/cdk_builder_utils.py

```diff
@@ -250,17 +251,59 @@ def get_queue_security_groups_full(managed_compute_security_group: ec2.CfnSecuri
     return queue_security_groups
 
 
+def add_cluster_iam_resource_prefix(stack_name, config, name: str, iam_type: str):
+    """Return a path and Name prefix from the Resource prefix config option."""
+    full_resource_path = None
+    full_resource_name = None
+    if config.iam and config.iam.resource_prefix:
+        iam_path, iam_name_prefix = get_path_n_name_prefix_from_iam_resource_prefix(config.iam.resource_prefix)
+        if iam_name_prefix:
+            if iam_type in ["AWS::IAM::Role", "AWS::IAM::InstanceProfile"]:
+                full_resource_name = iam_name_prefix + name + "-"
+                # Add a region for making the hash and Name Unique Globally
+                full_resource_name = (
+                    full_resource_name
+                    + hashlib.sha256((full_resource_name + stack_name + iam_type + config.region).encode("utf-8"))
+                    .hexdigest()[:12]
+                    .capitalize()
+                )
+
+            elif iam_type == "AWS::IAM::Policy":
+                full_resource_name = iam_name_prefix
+        if iam_path:
+            full_resource_path = iam_path
+
+    return full_resource_path, full_resource_name
+
+
 def add_lambda_cfn_role(scope, function_id: str, statements: List[iam.PolicyStatement], has_vpc_config: bool):
     """Return a CfnRole to be used for a Lambda function."""
+    role_id = f"{function_id}FunctionExecutionRole"
+    lambda_path = IAM_ROLE_PATH
+    policy_name = "LambdaPolicy"
+    # Need to reduce calls to this function
+    role_path, role_name = add_cluster_iam_resource_prefix(
+        scope.config.cluster_name, scope.config, name=f"{function_id}Role", iam_type="AWS::IAM::Role"
+    )
+    _, policy_name_prefix = add_cluster_iam_resource_prefix(
+        scope.config.cluster_name, scope.config, None, iam_type="AWS::IAM::Policy"
+    )
+    if role_path:
+        lambda_path = role_path
+    if policy_name_prefix:
+        policy_name = policy_name_prefix + "-" + policy_name
+    if role_name:
+        role_id = f"{function_id}Role"
```

Comment:
instead of reassigning the `role_id` we should only assign a value once with an if/else statement

Reply:
Assigning the value just once